internal/payloads/ami_payload.go

```diff
@@ -0,0 +1,48 @@
+package payloads
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/RHEnVision/provisioning-backend/internal/helpers"
+)
+
+type AmiResponse struct {
+	Distribution  string          `json:"distribution"`
+	ImageName     string          `json:"image_name"`
+	ImageRequests []ImageRequests `json:"image_requests"`
+}
+type Options struct {
+	ShareWithAccounts []string `json:"share_with_accounts"`
+}
+type UploadRequest struct {
+	Type    string  `json:"type"`
+	Options Options `json:"options"`
+}
+type ImageRequests struct {
+	Architecture  string        `json:"architecture"`
+	ImageType     string        `json:"image_type"`
+	UploadRequest UploadRequest `json:"upload_request"`
+}
+
+func CreateAMIrequest() string {
+	data := AmiResponse{
+		Distribution: "rhel-84",
+		ImageName:    "test-image",
+		ImageRequests: []ImageRequests{
+			{
+				Architecture: "x86_64",
+				ImageType:    "aws",
+				UploadRequest: UploadRequest{
+					Type: "aws",
+					Options: Options{
+						ShareWithAccounts: []string{os.Getenv("ACCOUNT_ID")},
+					},
+				},
+			},
+		},
+	}
+	payloadBytes, err := json.Marshal(data)
+	helpers.PanicOn(err)
+	return string(payloadBytes)
+}
```

Comment:
So `payloads` package was meant for our OpenAPI payloads only, this feel like this is a request into sources API? I think we should create a new package for this. I suggest `clients`, there you can create subfolder for each service. Looks like edge-api has this:

```
[lzap@nuc pkg]$ find clients/
clients/
clients/client.go
clients/clients_suite_test.go
clients/clients_test.go
clients/fdo
clients/fdo/client.go
clients/fdo/client_test.go
clients/fdo/fdo_suite_test.go
clients/imagebuilder
clients/imagebuilder/client.go
clients/imagebuilder/client_test.go
clients/imagebuilder/mock_imagebuilder
clients/imagebuilder/mock_imagebuilder/client.go
clients/inventory
clients/inventory/client.go
clients/inventory/mock_inventory
clients/inventory/mock_inventory/inventory.go
clients/playbookdispatcher
clients/playbookdispatcher/client.go
```

They have just a single go file per client, which is not good practice. They also have the same name `client.go` which makes navigation by typing file name harder. I would propose this:

* `internal/clients/sources/sources_client.go` - CRUD REST functions
* `internal/clients/sources/sources_client_test.go` - tests (edge-api use mocking)
* `internal/clients/sources/sources_payloads.go` - structs

Something like that.

Reply:
Great Idea! thanks. 
should I create one for AWS as well?